### PR TITLE
change release flow so that moderators needs to merge rather than che…

### PR DESCRIPTION
**Pull request by ptrovatelli**
…rry-picking for releasing

as discussed: cherry-picking while releasing takes a lot of time. Contributors should choose the appropriate branch upon PR creation so that the releasing will be easier for moderators.

### CODE REVIEW

File: BRANCHING-MODEL.md [hunk 1]
```diff
@@ -3,20 +3,25 @@ This section describes
...
+
+If in doubt please use dev branch.
+
 # Release and hotfix model
 ![Schemas](doc/branching_model.png)
 ## Releasing
-- Start a release/x.y.0 release branch off dev branch
-- Commit only bug fixes from `dev` branch onto this branch
+- Start an x.y.0 release branch off dev branch
+- Commit only bug fixes onto this branch (and merge those fixes back to dev)
```
Comment (valentijnscholten) [inline BRANCHING-MODEL.md:17]: this is not 100% clear. The release branch will need to be merged back into dev after the release. If bugfixes are needed in dev earlier, the whole release branch needs to be merged back into dev.

Comment (ptrovatelli) [reply BRANCHING-MODEL.md]: i've removed that text so that we can concentrate on the diagram where all the information is already

**Comment by ptrovatelli** [comment]
actually i haven't updated it yet after our last meeting. I need to double-check a couple of things :)

**Comment by ptrovatelli** [comment]
@DefectDojo/2-core-moderators  @DefectDojo/3-project-moderators  updated the flow after our last meeting. 
New requirements: 
- being able to release as fast as possible
  - release an "1.0.0" and then iterate on this for more bugfixes
- being able to ship any kind of fixes fast, without knowing the exact number in advance: 
  - commit bug fixes to release/1.0.x branch and release this whenever we wish. Keep hotfix flow to quickly release an urgent fix if release/1.0.x has merged PR but isn't ready to ship.

**Pull request merged**